deepmd/descriptor/se_a.py

```diff
@@ -738,7 +738,7 @@ def _filter_lower(
             # we can safely return the final xyz_scatter filled with zero directly
             return tf.cast(tf.fill((natom, 4, outputs_size[-1]), 0.), self.filter_precision)
           # natom x nei_type_i x out_size
-          xyz_scatter = tf.reshape(xyz_scatter, (-1, shape_i[1]//4, outputs_size[-1]))  
+          xyz_scatter = tf.cast(tf.reshape(xyz_scatter, (-1, shape_i[1]//4, outputs_size[-1])), self.filter_precision)  
```

Comment:
If I understand correctly, it causes the error that [`deepmd.utils.network.embedding_net`](https://github.com/deepmodeling/deepmd-kit/blob/82c787de0ea98e3c032368a5fd4ee46a1fac6283/deepmd/utils/network.py#L99) accepts an input with `self.filter_precision` precision but returns one with `self.mixed_prec`. It is inconsistent. You should both cast and cast back inside the `embedding_net` or outside that method, but do not cast inside the `embedding_net` and cast back here.